### PR TITLE
Addition of build files UT 

### PR DESCRIPTION
**Pull request by vineela-pachchipulusu**
Added new UT cases for build files
The total statements covered for the DM code is increased to 30.4%

### CODE REVIEW

**Comment by hugonicodemos** [comment]
@vineela-pachchipulusu Is this ready for review?

File: build/build_suite_test.go [hunk 1]
```diff
@@ -23,6 +29,134 @@ func TestBuild(t *testing.T) {
 	RunSpecs(t, "Build Suite")
 }
 
+var _ = BeforeSuite(func() {
+	th.SetupHTTP()
+	StartAPIHandlers()
+}, 60)
```
Comment (sselvara1) [inline build/build_suite_test.go:35]: will it be better to indent here?

File: build/build_suite_test.go [hunk 2]
```diff
@@ -23,6 +29,134 @@ func TestBuild(t *testing.T) {
...
+		}
+	default:
+		http.Error(w, `{"error": "Method not allowed"}`, http.StatusMethodNotAllowed)
+	}
+}
+
+func StartAPIHandlers() {
+
+	th.Mux.HandleFunc("/datanetworks", HandleResourceRequests)
+	th.Mux.HandleFunc("/networks", HandleResourceRequests)
```
Comment (sselvara1) [inline build/build_suite_test.go:153]: is this for platform networks?

File: build/build_suite_test.go [hunk 3]
```diff
@@ -362,4 +496,612 @@ var _ = Describe("Test Build utilities:", func() {
...
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "host",
+								Namespace: "default",
+							},
+						},
+					},
+				}
+
+				exp := `---
+apiVersion: v1
```
Comment (sselvara1) [inline build/build_suite_test.go:1016]: wondering if this can be indented for better readability.

File: build/build_suite_test.go [hunk 4]
```diff
@@ -23,6 +29,134 @@ func TestBuild(t *testing.T) {
...
+		}
+    ]
+}
+`
+const AddrPoolListBody = `
+{
+    "addrpools": [
+        {
+            "name": "management",
+			"uuid": "aa277c8e-7421-4721-ae6a-347771fe4fa6"
```
Comment (hugonicodemos) [inline build/build_suite_test.go:51]: over-indented

File: build/build_suite_test.go [hunk 5]
```diff
@@ -23,6 +29,134 @@ func TestBuild(t *testing.T) {
...
+const NetworkAddressPoolListBody = `
+{
+    "network_addresspools": [
+        {
+			"uuid": "11111111-a6e5-425e-9317-995da88d6694",
+			"network_uuid": "a48a7b6d-9cfa-24a4-8d48-f0e25d35984a",
+			"address_pool_uuid": "aa277c8e-7421-4721-ae6a-347771fe4fa6",
+			"network_name": "oam",
+			"address_pool_name": "oam-ipv4"
+		}
```
Comment (hugonicodemos) [inline build/build_suite_test.go:111]: over-indented

**Review by hugonicodemos — APPROVED** [review]
minor comments, but LGTM

**Review by sriram-gn — APPROVED** [review]

File: build/build_suite_test.go (hunk 2 above)
Comment (vineela-pachchipulusu) [reply build/build_suite_test.go:153]: yes it is for platformNetworks

File: build/build_suite_test.go (hunk 3 above)
Comment (vineela-pachchipulusu) [reply build/build_suite_test.go:1016]: It gives the yaml string with the mentioned indentation

File: build/build_suite_test.go (hunk 5 above)
Comment (vineela-pachchipulusu) [reply build/build_suite_test.go]: Updated

File: build/build_suite_test.go (hunk 4 above)
Comment (vineela-pachchipulusu) [reply build/build_suite_test.go]: updated

File: build/build_suite_test.go (hunk 1 above)
Comment (vineela-pachchipulusu) [reply build/build_suite_test.go:35]: Indentation is good enough here

**Review by sselvara1 — APPROVED** [review]

**Pull request merged**